### PR TITLE
Added bug report documentation

### PR DESCRIPTION
**Pull request by DavidLu1997**
Wrote effective bug report documentation with inspiration from:
 - http://www.chiark.greenend.org.uk/~sgtatham/bugs.html
 - http://www.mattermost.org/filing-issues/

Comments/suggestions/changes welcome

@it33 please review

### CODE REVIEW

**Comment by it33** [comment]
Really nice start, well written and mostly clear. 

Thoughts: 
1. Let's find the best possible samples--concise, clear, immediately actionable
2. Propose investing in getting this super short and effective. This is good, how can we make it great? 
3. Please review with JW and let's circle back on Core?

File: source/process/bug-report.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+# Bug Reports
+
+A bug for Mattermost is defined as "any obvious error". Errors on unsupported platforms are not considered bugs.
+
+Sample bugs:
+   - The front page is blank.
+   - /msg does not work for users outside of the current team.
+   - Whitespace is removed beside emoticons.
```
Comment (jwilander) [inline source/process/bug-report.md]: Link to jira tickets in samples

File: source/process/bug-report.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+   - @all does not work.
+
+## Bug Report Format
+
+A well-written bug report consists of several components:
+   1. **Summary**, the bug in one sentence.
+   2. **Reproduction Steps**, how to reproduce the bug.
+   3. **Expected Behavior**, describe the bug in detail.
+   4. **Observed Behavior**, what exactly happened.
+   5. **Possible Fixes** (optional), linking code/settings that may have caused the bug.
```
Comment (jwilander) [inline source/process/bug-report.md]: Screenshots (optional)

File: source/process/bug-report.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+Be as clear as you can in the summary, a good summary tells the reader:
+   - **What** happened? Be as exact as you can (e.g. The app crashes)
+   - **How** did it happen? Give a brief description of the process used (e.g. when switching channels from Town Square to any other channel)
+   - **Where** does it happen? Include the version number/platform. (e.g. on Android v5.1 running Mattermost v3.0.2.)
+   
+### Reproduction Steps
+
+State exactly which actions you took before encountering the bug. A good list of reproduction steps is:
+   - **Clear**, anyone should be able to reproduce the bug using just your instructions
+   - **Accurate**, no vague descriptions should be used. (e.g. "Switch channel to Off-Topic" vs. "Switch channel to Off-Topic using ALT+UP")
```
Comment (jwilander) [inline source/process/bug-report.md]: "Switch channel to Off-Topic using ALT+UP" maybe switch to "Switch channel to Off-Topic by clicking on the channel in the LHS" 0/5

File: source/process/bug-report.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+   - JavaScript errors (if applicable)
+   - Server logs (if applicable)
+
+### Possible Fixes
+
+Including any potential fixes, or areas to look into, for developers reading your bug report. Consider including section(s) of code.
+
+This section is completely optional.
+
+### Formatting
```
Comment (jwilander) [inline source/process/bug-report.md]: Remove formatting section

**Comment by DavidLu1997** [comment]
Changes made, @it33 give it a look?

**Comment by esethna** [comment]
Hey @DavidLu1997, going to close this for now. It's a good start and definitely a doc that we are interested in having!
From a quick glance I think we could find some better examples and and shorten it significantly. Definitely not urgent, but if you are interested in continuing to work on this, maybe try another iteration and then ask Lindy for a review? She helps file all the bug reports and will have a good idea of what it takes to make a great bug report.

**Pull request closed without merging**